components/merkledb/examples/migration.rs

```diff
@@ -290,3 +302,155 @@ fn main() {
+    {
+        let old_schema = v1::Schema::new(old_data.clone());
+        println!("Before migration:");
+        old_schema.print_wallets();
+    }
+    create_basic_migration(&fork);
+    db.merge(fork.into_patch()).unwrap();
+
+    check_and_finalize_migration(db.clone());
+}
+
+fn create_migration_with_helper(
+    helper: &mut MigrationHelper,
+    migrate_wallets_iter_loop: bool,
+) -> MdbResult<()> {
+    println!("\nStarted migration.");
+
+    {
+        let old_data = helper.old_data();
+        let new_data = helper.new_data();
+
+        let old_schema = v1::Schema::new(old_data);
+        let mut new_schema = v2::Schema::new(new_data);
+
+        // Move `ticker` and `divisibility` to `config`.
+        let config = v2::Config {
+            ticker: old_schema.ticker.get().unwrap(),
+            divisibility: old_schema.divisibility.get().unwrap_or(0),
+        };
+        new_schema.config.set(config);
+        // Mark these two indexes for removal.
+        new_data.create_tombstone("ticker");
+        new_data.create_tombstone("divisibility");
+    }
+
+    if migrate_wallets_iter_loop {
+        migrate_wallets_with_iter_loop(helper)?;
+    } else {
+        migrate_wallets_with_schema(helper.new_data(), helper.old_data());
+    }
+
+    Ok(())
+}
+
+/// Example of migration with `MigrationHelper`.
+fn migration_with_helper() {
+    println!("\n\nMigration with MigrationHelper.");
+
+    let db: Arc<dyn Database> = Arc::new(TemporaryDB::new());
+    let fork = db.fork();
+    create_initial_data(&fork);
+    fork.get_proof_list("unrelated.list").extend(vec![1, 2, 3]);
+    db.merge(fork.into_patch()).unwrap();
+
+    let mut helper = MigrationHelper::new(db.clone(), "test");
+    {
+        let old_schema = v1::Schema::new(helper.old_data().clone());
+        println!("Before migration:");
+        old_schema.print_wallets();
+    }
+
+    if let Err(err) = create_migration_with_helper(&mut helper, false) {
+        println!("Migration failed: {}", err);
+        return;
+    }
+
+    let new_state = helper.finish();
+    if let Err(err) = new_state {
+        println!("Migration finish failed: {}", err);
+        return;
+    }
+
+    check_and_finalize_migration(db.clone());
+}
+
+/// Example of migration with `MigrationHelper::iter_loop`.
+fn migration_with_helper_iter_loop() {
+    println!("\n\nMigration with MigrationHelper::iter_loop.");
+
+    let db: Arc<dyn Database> = Arc::new(TemporaryDB::new());
+    let fork = db.fork();
+    create_initial_data(&fork);
+    fork.get_proof_list("unrelated.list").extend(vec![1, 2, 3]);
+    db.merge(fork.into_patch()).unwrap();
+
+    let mut helper = MigrationHelper::new(db.clone(), "test");
+    {
+        let old_schema = v1::Schema::new(helper.old_data().clone());
+        println!("Before migration:");
+        old_schema.print_wallets();
+    }
+
+    if let Err(err) = create_migration_with_helper(&mut helper, true) {
```

Comment:
Nit: It could make sense `migration_with_helper_iter_loop` to return `exonum_merkledb::Result<()>`, so that this could be shortened to

```rust
create_migration_with_helper(&mut helper, true)?;
```